### PR TITLE
Add type casting

### PR DESCRIPTION
**Pull request by ryansch**
## Typecasting/Coersion
You can supply Metaractor with a callable that will typecast incoming parameters:
```ruby
optional :needs_to_be_a_string, type: ->(value) { value.to_s }
```

You can also configure Metaractor with named types and use them:
```ruby
Metaractor.configure do |config|
  config.register_type(:boolean, ->(value) { ActiveModel::Type::Boolean.new.cast(value) })
end
```
```ruby
required :is_awesome, type: :boolean
```

**Note**: Typecasters will _not_ be called on `nil` values.

### CI!
This PR also adds CI via buildkite.

### CODE REVIEW

File: spec/metaractor_spec.rb [hunk 1]
```diff
@@ -322,6 +322,77 @@ def call
...
+        expect(result.job_title).to eq true
+        expect(result.job_title_required).to eq false
+      end
+
+      context 'with allow_blank' do
+        let(:types_class) do
+          Class.new do
+            include Metaractor
+
+            optional :boom,
```
Comment (ryansch) [inline spec/metaractor_spec.rb:377]: @morganick Is this what you had in mind?

Comment (morganick) [reply spec/metaractor_spec.rb:377]: 👨‍🍳 💋

File: spec/metaractor_spec.rb [hunk 2]
```diff
@@ -322,6 +322,77 @@ def call
...
+              allow_blank: true,
+              type: ->(value) { raise "BOOM" if value.nil? }
```
Comment (morganick) [inline spec/metaractor_spec.rb:379]:
```ruby
optional :pow, type: ->(value) { raise "POW" if value.nil? }
```
to exercise the other part of the conditional? Expecting that result doesn't even have the `pow` key.

**Review by morganick — APPROVED** [review]
🎉 🙌

**Pull request merged**